algorithms/kernel/k_nearest_neighbors/oneapi/bf_knn_classification_predict_kernel_ucapi_impl.i

```diff
@@ -144,145 +114,119 @@ Status KNNClassificationPredictKernelUCAPI<algorithmFpType>::
 
-    BlockDescriptor<algorithmFpType> probeRows;
-    DAAL_CHECK_STATUS_VAR(ntData->getBlockOfRows(0, nProbeRows, readOnly, probeRows));
-    auto probes_all = probeRows.getBuffer();
-
-    BlockDescriptor<algorithmFpType> dataRows;
-    DAAL_CHECK_STATUS_VAR(points->getBlockOfRows(0, nDataRows, readOnly, dataRows));
-    auto data_all = dataRows.getBuffer();
-
-    BlockDescriptor<int> labelRows;
-    DAAL_CHECK_STATUS_VAR(labels->getBlockOfRows(0, nDataRows, readOnly, labelRows));
-    auto labels_all = labelRows.getBuffer();
+    SelectIndexed::Params params(nK, TypeIds::id<algorithmFpType>(), dataBlockSize, parameter->engine);
+    SelectIndexedFactory factory;
+    SharedPtr<SelectIndexed> selector(factory.Create(nK, params, &st));
+    DAAL_CHECK_STATUS_VAR(st);
 
     for (size_t pblock = 0; pblock < nProbeBlocks; pblock++)
     {
         const size_t pfirst = pblock * probeBlockSize;
-        size_t plast = pfirst + probeBlockSize;
+        size_t plast        = pfirst + probeBlockSize;
 
         if (plast > nProbeRows)
-            plast = nProbeRows;
-        const size_t curProbeBlockSize = plast - pfirst;
         {
-            DAAL_ITTNOTIFY_SCOPED_TASK(compute.copy_query);
-            context.copy(probe_temp, 0, probes_all, pfirst * nFeatures, nFeatures * curProbeBlockSize, &st);
+            plast = nProbeRows;
         }
-        auto probes = probe_temp.template get<algorithmFpType>();
-        uint32_t partCount = 0;
+        const size_t curProbeBlockSize = plast - pfirst;
 
+        BlockDescriptor<algorithmFpType> probeRows;
+        DAAL_CHECK_STATUS_VAR(ntData->getBlockOfRows(pfirst, curProbeBlockSize, readOnly, probeRows));
+        auto curProbes = probeRows.getBuffer();
+
+        uint32_t partCount = 0;
         for (size_t dblock = 0; dblock < nDataBlocks; dblock++)
         {
             const size_t dfirst = dblock * dataBlockSize;
-            size_t dlast = dfirst + dataBlockSize;
+            size_t dlast        = dfirst + dataBlockSize;
 
             if (dlast > nDataRows)
+            {
                 dlast = nDataRows;
+            }
             const size_t curDataBlockSize = dlast - dfirst;
 
+            BlockDescriptor<int> labelRows;
+            DAAL_CHECK_STATUS_VAR(labels->getBlockOfRows(dfirst, dataBlockSize, readOnly, labelRows));
+            auto dataLabels = labelRows.getBuffer();
+            BlockDescriptor<algorithmFpType> dataRows;
+            DAAL_CHECK_STATUS_VAR(points->getBlockOfRows(dfirst, curDataBlockSize, readOnly, dataRows));
+            auto curData = dataRows.getBuffer();
+
+            // Collect sums of squares from train data
+            auto sumResult = math::SumReducer::sum(math::Layout::RowMajor, curData, curDataBlockSize, nFeatures, &st);
+            DAAL_CHECK_STATUS_VAR(st);
+            // Initialize GEMM distances
+            initDistances(context, init_distances, sumResult.sumOfSquares, distances, curDataBlockSize, curProbeBlockSize, &st);
```

Comment:
It is totally unclear which of the parameters are input and which are output with such a big amount of them. If you can not return outputs like it is done in `math::SumReducer` for performance reasons, please, follow the convention:
```cpp
services::Status func(input_1, input_2, ..., output_1, ... output_k);
```